### ISSUE TITLE
Binding UI not generating actions or allowing changes

### ISSUE THREAD

**Issue by Rhybo** [ValveSoftware/steamvr_unity_plugin#312]
I seem to be having an issue as referenced in #197 . 

My current setup:

- Unity 2018.1f1
- SteamVR 1.2.10 (1548804153) but see my update below
- HTC Vive Pro

I deleted the default action sets and just tried to Save and Generate my own action set, which according to Unity was successful. However, it doesn't register as a valid action set as Unity throws a NullReferenceException since it can't see my controllers. My project was working until I deleted the default action set. Now I cannot use the controllers at all.

I have tried both starting SteamVR through Steam and letting Unity start SteamVR, both with the same result; the bindings UI page opens up and the edit button for the controllers only makes the screen darken for multiple seconds before lightening back up. I have tried opening the binding UI in FireFox and Chrome for my Unity project and both have the same result.

I deleted the SteamVR_Input folder and tried to regenerate default bindings.  Even though they appeared in the SteamVR Input window after deletion, Save and Generate did nothing and now I can't get my controllers to show up in the Unity project (I can see them however if push the Steam menu button on the controller).

Edit and Update: I'm a dummy, my actions.json file now included.  To try to isolate this issue I downloaded SteamVR beta v.1.3.1 (1549469152) and created a brand new project and re-imported assets.  I'm able to work on my project again but only with the default bindings.  The same thing happens when I try to change bindings after opening the binding UI from within my Unity project, however, if I select Unity.exe (or any other application) instead of my project name I can adjust bindings.

```
{
  "actions": [
    {
      "name": "/actions/default/in/InteractUI",
      "type": "boolean"
    },
    {
      "name": "/actions/default/in/Teleport",
      "type": "boolean"
    },
    {
      "name": "/actions/default/in/GrabPinch",
      "type": "boolean"
    },
    {
      "name": "/actions/default/in/GrabGrip",
      "type": "boolean"
    },
    {
      "name": "/actions/default/in/Pose",
      "type": "pose"
    },
    {
      "name": "/actions/default/in/SkeletonLeftHand",
      "type": "skeleton",
      "skeleton": "/skeleton/hand/left"
    },
    {
      "name": "/actions/default/in/SkeletonRightHand",
      "type": "skeleton",
      "skeleton": "/skeleton/hand/right"
    },
    {
      "name": "/actions/default/in/Squeeze",
      "type": "vector1",
      "requirement": "optional"
    },
    {
      "name": "/actions/default/in/HeadsetOnHead",
      "type": "boolean",
      "requirement": "optional"
    },
    {
      "name": "/actions/default/out/Haptic",
      "type": "vibration"
    },
    {
      "name": "/actions/platformer/in/Move",
      "type": "vector2"
    },
    {
      "name": "/actions/platformer/in/Jump",
      "type": "boolean"
    },
    {
      "name": "/actions/buggy/in/Steering",
      "type": "vector2"
    },
    {
      "name": "/actions/buggy/in/Throttle",
      "type": "vector1"
    },
    {
      "name": "/actions/buggy/in/Brake",
      "type": "boolean"
    },
    {
      "name": "/actions/buggy/in/Reset",
      "type": "boolean"
    },
    {
      "name": "/actions/mixedreality/in/ExternalCamera",
      "type": "pose",
      "requirement": "optional"
    },
    {
      "name": "/actions/mtx/in/GrabAsset",
      "type": "boolean"
    },
    {
      "name": "/actions/mtx/in/Fly",
      "type": "boolean"
    }
  ],
  "action_sets": [
    {
      "name": "/actions/default",
      "usage": "single"
    },
    {
      "name": "/actions/platformer",
      "usage": "single"
    },
    {
      "name": "/actions/buggy",
      "usage": "single"
    },
    {
      "name": "/actions/mixedreality",
      "usage": "single"
    },
    {
      "name": "/actions/mtx",
      "usage": "single"
    }
  ],
  "default_bindings": [
    {
      "controller_type": "vive_controller",
      "binding_url": "bindings_vive_controller.json"
    },
    {
      "controller_type": "oculus_touch",
      "binding_url": "bindings_oculus_touch.json"
    },
    {
      "controller_type": "knuckles",
      "binding_url": "bindings_knuckles.json"
    },
    {
      "controller_type": "holographic_controller",
      "binding_url": "bindings_holographic_controller.json"
    },
    {
      "controller_type": "vive",
      "binding_url": "binding_vive.json"
    },
    {
      "controller_type": "vive_pro",
      "binding_url": "binding_vive_pro.json"
    },
    {
      "controller_type": "rift",
      "binding_url": "binding_rift.json"
    },
    {
      "controller_type": "holographic_hmd",
      "binding_url": "binding_holographic_hmd.json"
    },
    {
      "controller_type": "vive_tracker_camera",
      "binding_url": "binding_vive_tracker_camera.json"
    }
  ],
  "localization": [
    {
      "language_tag": "en_US",
      "/actions/default/in/GrabGrip": "Grab Grip",
      "/actions/default/in/GrabPinch": "Grab Pinch",
      "/actions/default/in/HeadsetOnHead": "Headset on head (proximity sensor)",
      "/actions/default/in/InteractUI": "Interact With UI",
      "/actions/default/in/Pose": "Pose",
      "/actions/default/in/SkeletonLeftHand": "Skeleton (Left)",
      "/actions/default/in/SkeletonRightHand": "Skeleton (Right)",
      "/actions/default/in/Teleport": "Teleport",
      "/actions/default/out/Haptic": "Haptic",
      "/actions/platformer/in/Jump": "Jump",
      "/actions/mtx/in/GrabAsset": "Grab Asset"
      "/actions/mtx/in/Fly": "Fly"
    }
  ]
}
```

**Comment by zite** [comment]
If you delete the default actionset you'll need to make a Pose action that the controllers can use to get their orientation. That's how we're tracking controllers now. Would that solve your issue? Are you saying you're having issues opening the binding UI? The binding ui is different for each game, and will be different if you're working in the editor or working on a built version of the game.

**Comment by Rhybo** [comment]
The trash controllers are no longer in if you since I started a brand new project.

I am trying to assign bindings based on my project in the editor, I don't have a built version.  The binding UI, which had my Unity project folder name up top, doesn't do anything when I click "Edit".  The screen darkens for about 5 seconds then returns to it's original state.

**Comment by zite** [comment]
Hmm, if you can post the binding file for the controllers your trying to edit that might help. If it's too big for here there's https://gist.github.com/ too.

**Comment by Rhybo** [comment]
@keithbradner Thanks for holding my hand through the process, here's my [Steam_VR_Input_BindingFile.cs](https://gist.github.com/Rhybo/34d8a52be8d56c587e7d02380d9af8d7).  

If that's not what you're looking for, here's the binding_vive_pro.json file in my project:

```
{
   "alias_info" : {},
   "bindings" : {
      "/actions/default" : {
         "chords" : [],
         "haptics" : [],
         "poses" : [],
         "skeleton" : [],
         "sources" : [
            {
               "inputs" : {
                  "click" : {
                     "output" : "/actions/default/in/headsetonhead"
                  }
               },
               "mode" : "button",
               "path" : "/user/head/proximity"
            }
         ]
      }
   },
   "controller_type" : "vive_pro",
   "description" : "",
   "name" : "vive_pro defaults",
   "options" : {},
   "simulated_actions" : []
}
```

On a side note, I entered play mode in my project and opened the binding UI through the SteamVR Input tab as referenced in the SteamVR blog post [here](https://steamcommunity.com/sharedfiles/filedetails/?id=1416820276) but it still isn't working.

**Comment by bedaites** [comment]
I'm experiencing quite a similar issue; whereby my binding UI will not open, either through the SteamVR Input tab (Chrome says "local host didn't send any data") or through the SteamVR window itself (opens and loads blank). Worth posting my own issue over? Any solutions? 

I'm running the latest version of SteamVR plugin, SteamVR Beta and Unity 5.5.4p4.

**Comment by zite** [comment]
Can you both make steamvr system reports and send them to me?
SteamVR app -> Menu -> Create System Report

keithb@valvesoftware.com

**Comment by Rhybo** [comment]
Update:  I was able to pull up the binding UI on my home network with the same project.  When I opened up my project on the same laptop on my school network the same problem persisted, however, since I set up and saved bindings at home and saved them, I was able to use them here on campus as long as I have Steam open.  

My problem is mostly solved as a result, however, it seems prohibitive that I cannot use bindings generated for a Unity project unless my Steam client is open.  As of now, if I only have SteamVR open (initialized by opening my Unity project) but not Steam, I can't use my controllers at all unless I hit the Vive menu button to pull up the SteamVR UI.  I just confirmed by hitting play on the Unity editor both while logged in to Steam and while not.  Is there a way to generate and save bindings locally without having network access to Steam?
